direct/core/src/test/java/cz/o2/proxima/direct/storage/InMemStorageTest.java

```diff
@@ -571,4 +572,81 @@ public boolean onNext(StreamElement ingest, OnNextContext context) {
     successObserverAllReceived.await();
     assertEquals(1, failingObserverMessages.get());
   }
+
+  @Test
+  public void testObserveMultiplePartitions() throws InterruptedException {
```

Comment:
This tests CommitLogReader only, we should test RandomAccessReader as well.

Reply:
RandomAccessReader is out of scope of this PR, I'll add checks so that missing support does not bite us in the future